src/pages/InterProtocol.tsx

```diff
@@ -76,43 +80,61 @@ export function InterProtocol() {
     {},
   );
 
-  const boardAuxes: { [key: string]: number } = dashboardResponse.boardAuxes.nodes.reduce(
-    (agg, node) => ({ ...agg, [node.allegedName]: node.decimalPlaces }),
+  const boardAuxes: { [key: string]: number } = dashboardResponse?.boardAuxes?.nodes?.reduce(
+    (agg, node) => ({ ...agg, [node?.allegedName]: node?.decimalPlaces }),
     {},
   );
 
+  const istTokenDivisor = getTokenDivisor(boardAuxes, 'IST');
+
   const psmMinted =
-  dashboardResponse.psmMetrics.nodes.reduce((agg, node) => agg + Number(node.mintedPoolBalance), 0) / 1_000_000;
-  const psmAnchor = dashboardResponse.psmMetrics.nodes.reduce(
-    (agg, node) => agg + Number(node.anchorPoolBalance) / 10 ** (boardAuxes[node.denom] || 6),
-    0,
-  );
-  const vaultMinted =
-  dashboardResponse.vaultManagerMetrics.nodes.reduce((agg, node) => agg + Number(node.totalDebt), 0) / 1_000_000;
+    dashboardResponse?.psmMetrics?.nodes?.reduce((agg, node) => agg + Number(node?.mintedPoolBalance), 0) /
+    istTokenDivisor;
+
+  const psmAnchor = dashboardResponse.psmMetrics.nodes.reduce((agg, node) => {
+    const decimalPlaces = (boardAuxes && boardAuxes[node?.denom]) || 6;
+    const anchorPoolBalance = Number(node?.anchorPoolBalance) / 10 ** decimalPlaces;
+    return agg + anchorPoolBalance;
+  }, 0);
+
+  const vaultMinted = dashboardResponse.vaultManagerMetrics.nodes.reduce((agg, node) => {
+    const tokenDivisor = getTokenDivisor(boardAuxes, node?.liquidatingCollateralBrand);
+    return agg + Number(node?.totalDebt) / tokenDivisor;
+  }, 0);
+
   const totalMinted = psmMinted + vaultMinted;
 
   const vaultMintLimit =
-  dashboardResponse.vaultManagerGovernances.nodes.reduce((agg, node) => agg + Number(node.debtLimit), 0) / 1_000_000;
-  const psmMintLimit = dashboardResponse.psmGovernances.nodes.reduce((agg, node) => agg + Number(node.mintLimit), 0) / 1_000_000;
+    dashboardResponse?.vaultManagerGovernances?.nodes?.reduce((agg, node) => agg + Number(node?.debtLimit), 0) /
+    istTokenDivisor;
+
+  const psmMintLimit =
+    dashboardResponse?.psmGovernances?.nodes?.reduce((agg, node) => agg + Number(node?.mintLimit), 0) / istTokenDivisor;
+
   const totalMintLimit = vaultMintLimit + psmMintLimit;
 
   // bottom cards
   const totalReserve = dashboardResponse.reserveMetrics.nodes.reduce(
     (agg, node) =>
       agg +
-      node.allocations.nodes.reduce((agg_, node_) => {
+      node?.allocations?.nodes.reduce((agg_, node_) => {
+        const tokenDivisor = getTokenDivisor(boardAuxes, node_?.denom);
         const allocationInUsd =
-          ((Number(node_.value) / 1_000_000) * Number(oraclePrices[node_.denom]?.typeOutAmount || 1_000_000)) / 1_000_000;
+          ((Number(node_?.value) / tokenDivisor) * Number(oraclePrices[node_?.denom]?.typeOutAmount || tokenDivisor)) /
+          tokenDivisor;
```

Comment:
```suggestion
          ((Number(node_?.value) / tokenDivisor) * Number(oraclePrices[node_?.denom]?.typeOutAmount || 0)) /
          Number(oraclePrices[node_?.denom]?.typeInAmount;
```

Reply:
we should probably use `Number(oraclePrices[node_?.denom]?.typeInAmount` instead of the deafault of 0 then

Reply:
> we should probably use Number(oraclePrices[node_?.denom]?.typeInAmount

I reckon you are impying `typeOutAmount`?

Reply:
no `typeInAmount`

the goal is to divide `typeInAmount` by `typeInAmount` to create a value of 1

Reply:
but i think if `typeOutAmount` is not present, then `typeInAmount` wont also be present. lets do this instead:
```
Number(oraclePrices[node_?.denom]?.typeOutAmount || 1)) /
          Number(oraclePrices[node_?.denom]?.typeInAmount || 1);
```